debezium/src/main/scala/akka/stream/alpakka/debezium/DebeziumSettings.scala

```diff
@@ -0,0 +1,124 @@
+
+import java.util
+import java.util.Properties
+
+import io.debezium.engine.spi.OffsetCommitPolicy
+import org.apache.kafka.connect.connector.Connector
+import org.apache.kafka.connect.storage.{Converter, OffsetBackingStore}
+
+import scala.concurrent.duration.FiniteDuration
+import scala.collection.JavaConverters._
+
+final class DebeziumSettings private (
+    val name: String,
+    val connectorClass: Class[_ <: Connector],
+    val offsetStorage: Class[_ <: OffsetBackingStore],
+    val offsetStorageFileName: String,
+    val offsetCommitPolicy: Class[_ <: OffsetCommitPolicy],
+    val offsetFlushInterval: FiniteDuration,
+    val offsetFlushTimeout: FiniteDuration,
+    val internalKeyConverter: Class[_ <: Converter],
+    val internalValueConverter: Class[_ <: Converter],
+    val customSettings: Map[String, Any],
+    val maxInMemoryElements: Int
+) {
+
+  def copy(
+      name: String = name,
+      connectorClass: Class[_ <: Connector] = connectorClass,
+      offsetStorage: Class[_ <: OffsetBackingStore] = offsetStorage,
+      offsetStorageFileName: String = offsetStorageFileName,
+      offsetCommitPolicy: Class[_ <: OffsetCommitPolicy] = offsetCommitPolicy,
+      offsetFlushInterval: FiniteDuration = offsetFlushInterval,
+      offsetFlushTimeout: FiniteDuration = offsetFlushTimeout,
+      internalKeyConverter: Class[_ <: Converter] = internalKeyConverter,
+      internalValueConverter: Class[_ <: Converter] = internalValueConverter,
+      customSettings: Map[String, Any] = customSettings,
+      maxInMemoryElements: Int = maxInMemoryElements
+  ): DebeziumSettings =
+    new DebeziumSettings(name,
+                         connectorClass,
+                         offsetStorage,
+                         offsetStorageFileName,
+                         offsetCommitPolicy,
+                         offsetFlushInterval,
+                         offsetFlushTimeout,
+                         internalKeyConverter,
+                         internalValueConverter,
+                         customSettings,
+                         maxInMemoryElements)
+
+  def properties: Properties = {
+    val properties = new Properties()
+
+    properties.put("name", name)
+    properties.put("connector.class", connectorClass.getName)
+    properties.put("offset.storage", offsetStorage)
+    properties.put("offset.storage.file.filename", offsetStorageFileName)
+    properties.put("offset.commit.policy", offsetCommitPolicy.getName)
+    properties.put("offset.flush.interval.ms", s"${offsetFlushInterval.toMillis}")
+    properties.put("offset.flush.timeout.ms", s"${offsetFlushTimeout.toMillis}")
+    properties.put("internal.key.converter", internalKeyConverter.getName)
+    properties.put("internal.value.converter", internalValueConverter.getName)
+
+    val custom: util.Map[Object, Object] = customSettings.asJava.asInstanceOf[util.Map[Object, Object]]
+    properties.putAll(custom)
+
+    properties
+  }
+}
+
+object DebeziumSettings {
```

Comment:
We'll need to elaborate this to include defaults in a `reference.conf` and a fluent API for Scala and Java users to call at runtime.